### PR TITLE
feat(stream): append-only TopN with ties

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
close https://github.com/risingwavelabs/risingwave/issues/5642
## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Comment by xxchan** [comment]
cc @BugenZhao PTAL

File: e2e_test/streaming/append_only.slt [hunk 1]
```diff
@@ -1,3 +1,6 @@
+statement ok
+SET RW_IMPLICIT_FLUSH TO true;
```
Comment (BugenZhao) [inline e2e_test/streaming/append_only.slt:2]: Also remove existing `FLUSH`es?

File: src/stream/src/executor/top_n/top_n_cache.rs [hunk 2]
```diff
@@ -468,3 +476,171 @@ impl TopNCacheTrait for TopNCache<true> {
...
+        let middle_last_order_by = &middle_last.0 .0.clone();
+
+        match sort_key.cmp(middle_last_order_by) {
+            Ordering::Less => {
+                // The row is in middle.
+                let num_ties = self
+                    .middle
+                    .range((middle_last_order_by.clone(), vec![])..)
+                    .count();
+                // We evict the last row and its ties only if the number of remaining rows still is
```
Comment (BugenZhao) [inline src/stream/src/executor/top_n/top_n_cache.rs:604]:
```suggestion
                // We evict the last row and its ties only if the number of remaining rows is
```

File: src/stream/src/executor/top_n/top_n_cache.rs [hunk 3]
```diff
@@ -468,3 +476,171 @@ impl TopNCacheTrait for TopNCache<true> {
...
+                // still larger than limit, i.e., there are limit-1 other rows.
+                //
+                // e.g., limit = 3, [1,1,1,1]
+                // insert 0 -> [0,1,1,1,1]
+                // insert 0 -> [0,0,1,1,1,1]
+                // insert 0 -> [0,0,0]
+                if self.middle.len() - num_ties + 1 >= self.limit {
+                    while let Some(middle_last) = self.middle.last_entry()
+                    && middle_last.key().0 == middle_last_order_by.clone() {
```
Comment (BugenZhao) [inline src/stream/src/executor/top_n/top_n_cache.rs:613]: No need to `clone` here?

**Review by BugenZhao — COMMENTED** [review]
LGTM

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**